omod/src/main/java/org/openmrs/module/atomfeed/web/controller/AtomfeedConfigurationController.java

```
@@ -0,0 +1,137 @@
+
+	@Autowired
+	private FeedConfigurationService feedConfigurationService;
+
+	/**
+	 * Sets the UI model attributes used to display the atomfeed configuration etc.
+	 *
+	 * @param model injected the page model object
+	 * @param success injected the flag used to choose the type of alert message
+	 * @param alertMessage injected the message used to display an alert message (display the alert if the value isn't null)
+	 * @return the redirect URL to atomfeedConfiguration.jsp
+	 */
+	@RequestMapping("/configuration")
+	public String get(ModelMap model,
+			@RequestParam(value = AtomfeedMessageUtils.SUCCESS_MESSAGE, required = false) boolean success,
+			@RequestParam(value = AtomfeedMessageUtils.ALERT_MESSAGE_MODEL, required = false) String alertMessage) {
+		String configuration =
+				AtomfeedUtils.writeFeedConfigurationToJsonString(feedConfigurationService.getGeneralConfiguration());
+		model.addAttribute(CONFIGURATION_MODEL, configuration);
+		model.addAttribute(AtomfeedMessageUtils.SUCCESS_MESSAGE, success);
+		model.addAttribute(AtomfeedMessageUtils.ALERT_MESSAGE_MODEL, alertMessage);
+		return "/module/atomfeed/atomfeedConfiguration";
+	}
+
+	/**
+	 * Saves the configuration (sent as JSON string) into the server.
+	 * Notifies the user about the result of this operation.
+	 *
+	 * @param model injected the page model object
+	 * @param json injected the JSON representation of the configuration
+	 * @return the redirect URL (if success to /module/atomfeed/atomfeed if not then /module/atomfeed/configuration)
+	 */
+	@RequestMapping(value = "/saveConfiguration", method = RequestMethod.POST)
+	public String post(ModelMap model,
+			@RequestParam("json") String json) {
+		try {
+			feedConfigurationService.saveConfig(json);
+			AtomfeedMessageUtils.successMessage(model, SAVE_CONFIG_SUCCESS);
+			return "redirect:/module/atomfeed/atomfeed.form";
+		} catch (Exception e) {
```

can you use less generic exception ? because here you try to catch all exceptions, even NPE

---

Good point, thanks.